lib/app_worker.js

```diff
@@ -34,14 +34,28 @@ function startServer(err) {
   app.removeListener('startTimeout', startTimeoutHandler);
 
   let server;
+  let httpsOptions;
   if (options.https) {
-    const httpsOptions = Object.assign({}, options.https, {
+    httpsOptions = Object.assign({}, options.https, {
       key: fs.readFileSync(options.https.key),
       cert: fs.readFileSync(options.https.cert),
     });
-    server = require('https').createServer(httpsOptions, app.callback());
+  }
+
+  if (options.http2) {
```

Comment:
Need testcase

Reply:
Ok, I will add test cases later. This may be a bit complicated as `supertest` does not support http2.